server/src/main/java/io/crate/action/sql/Sessions.java

```diff
@@ -70,6 +71,10 @@ public class Sessions {
         Setting.Property.Exposed
     );
 
+    public static final Setting<Integer> MEMORY_LIMIT = Setting.intSetting(
```

Comment:
Any reason not to use a `byteSizeSetting()`? This would be more comfortable especially with larger values (1g).

Reply:
Mostly because there's a bit of a mismatch between the `Setting` and `SessionSetting` classes. The former have specialized versions like `byteSizeSetting` and `SessionSetting` doesn't.

I made a note to look into unifying this a bit more to have consistent behavior between the two. But planned to do that later, maybe during the feature freeze